injected/src/features/password-import.js

```diff
@@ -0,0 +1,233 @@
+            return settingsButton
+        }
+        return await withExponentialBackoff(fn)
+    }
+
+    /**
+     * @returns {Promise<HTMLElement|Element|null>}
+     */
+    async findSignInButton () {
+        return await withExponentialBackoff(() => document.querySelector(this.signinButtonSelector))
+    }
+
+    /**
+     * Checks if the path is supported and animates/taps the element if it is.
+     * @param {string} path
+     */
+    async handleElementForPath (path) {
+        // FIXME: we need to check if the path is supported, otherwise
+        // for some reason google doesn't wait to proceed with the signin step.
+        // Not too sure why this is happening, there are no errors on the console.
+
+        if ([this.#exportButtonSettings.path, this.#settingsButtonSettings.path, this.#signInButtonSettings.path].indexOf(path) !== -1) {
+            try {
+                const { element, style, shouldTap } = await this.getElementAndStyleFromPath(path) ?? {}
+                if (element != null) {
+                    shouldTap ? this.autotapElement(element) : this.animateElement(element, style)
+                }
+            } catch {
+                console.error('password-import: handleElementForPath failed for path:', path)
+            }
+        }
+    }
+
+    /**
+     * @returns {string}
+     */
+    get exportButtonContainerSelector () {
+        return this.#exportButtonSettings?.selectors?.join(',')
+    }
+
+    /**
+     * @returns {string}
+     */
+    get exportButtonLabelTextSelector () {
+        return this.#exportButtonSettings?.labelTexts
+            .map(text => `button[aria-label="${text}"]`)
+            .join(',')
+    }
+
+    /**
+     * @returns {string}
+     */
+    get signinLabelTextSelector () {
+        return this.#signInButtonSettings?.labelTexts
+            .map(text => `a[aria-label="${text}"]:not([target="_top"])`)
+            .join(',')
+    }
+
+    /**
+     * @returns {string}
+     */
+    get signinButtonSelector () {
+        return `${this.#signInButtonSettings?.selectors?.join(',')}, ${this.signinLabelTextSelector}`
+    }
+
+    /**
+     * @returns {string}
+     */
+    get settingsLabelTextSelector () {
+        return this.#settingsButtonSettings?.labelTexts
+            .map(text => `a[aria-label="${text}"]`)
+            .join(',')
+    }
+
+    /**
+     * @returns {string}
+     */
+    get settingsButtonSelector () {
+        return `${this.#settingsButtonSettings?.selectors?.join(',')}, ${this.settingsLabelTextSelector}`
+    }
+
+    /**
+     * @param {any} settings
+     */
+    setButtonSettings (settings) {
+        this.#exportButtonSettings = settings?.exportButton
+        this.#settingsButtonSettings = settings?.settingsButton
+        this.#signInButtonSettings = settings?.signInButton
+    }
+
+    init (args) {
+        this.setButtonSettings(args?.featureSettings?.passwordImport || {})
```

Comment:
There should be a getter for settings to use instead of this argument. We're trying to deprecate this args usuage.

Reply:
Done!